File: dataset/output/MISTRAL_codestral-latest/Task148_MISTRAL_codestral-latest.c
#include <stdio.h>
#include <stdlib.h>

#define MAX_SIZE 100

typedef struct {
    int items[MAX_SIZE];
    int top;
} Stack;

void init(Stack *stack) {
    stack->top = -1;
}

void push(Stack *stack, int item) {
    if (stack->top >= MAX_SIZE - 1) {
        printf("Stack overflow
");
        return;
    }
    stack->items[++stack->top] = item;
}

int pop(Stack *stack) {
    if (stack->top < 0) {
        printf("Stack underflow
");
        exit(EXIT_FAILURE);
    }
    return stack->items[stack->top--];
}

int peek(Stack *stack) {
    if (stack->top < 0) {
        printf("Stack is empty
");
        exit(EXIT_FAILURE);
    }
    return stack->items[stack->top];
}

File: dataset/output/MISTRAL_codestral-latest/Task148_MISTRAL_codestral-latest.java
import java.util.EmptyStackException;
import java.util.Stack;

public class MyStack {
    private Stack<Integer> stack;

    public MyStack() {
        stack = new Stack<>();
    }

    public void push(int item) {
        stack.push(item);
    }

    public int pop() {
        if (stack.isEmpty()) {
            throw new EmptyStackException();
        }
        return stack.pop();
    }

    public int peek() {
        if (stack.isEmpty()) {
            throw new EmptyStackException();
        }
        return stack.peek();
    }
}